### PR TITLE
Support percentage tolerance for lexicographic optimization

### PR DESCRIPTION
**Pull request by skzhang1**
## Why are these changes needed?

Support percentage tolerance for lexicographic tuning

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/tune/test_lexiflow.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def _BraninCurrin(config):
     return {"brain": brain_result, "currin": currin_result}
 
 
-def test_lexiflow():
+def test_lexiflow(mode="absolute"):
```
Comment (sonichi) [inline test/tune/test_lexiflow.py:35]: Add a test for (mode="percentage"). pytest will look for test function and won't execute 194-196.

File: flaml/tune/searcher/flow2.py [hunk 2]
```diff
@@ -417,47 +419,31 @@ def lexico_compare(self, result) -> bool:
...
-                                        k_metric
-                                    ].replace("%", "")
-                                )
-                            ),
-                            k_target,
-                        ]
+                        )
                     )
-                ) and (
+                if (result[k_metric] < max([tolerance_bound, k_target])) and (
```
Comment (sonichi) [inline flaml/tune/searcher/flow2.py:442]:
```suggestion
                if result[k_metric] < max(tolerance_bound, k_target) and (
```

File: flaml/tune/searcher/flow2.py [hunk 3]
```diff
@@ -417,47 +419,31 @@ def lexico_compare(self, result) -> bool:
...
-                                + 0.01
-                                * float(
-                                    self.lexico_objectives["tolerances"][
-                                        k_metric
-                                    ].replace("%", "")
-                                )
-                            ),
+                            tolerance_bound,
                             k_target,
                         ]
```
Comment (sonichi) [inline flaml/tune/searcher/flow2.py:448]: remove unnecessary []

File: test/tune/test_lexiflow.py [hunk 4]
```diff
@@ -132,7 +126,27 @@ def evaluate_function(configuration):
...
+    )
+    print(analysis.best_trial)
+    print(analysis.best_config)
+    print(analysis.best_result)
+
     # lexico tune
+    lexico_objectives["targets"] = {"error_rate": 0.0, "flops": 0.0}
+    lexico_objectives["modes"] = ["min", "min"]
+
+    # 1. lexico tune: absoute tune
```
Comment (sonichi) [inline test/tune/test_lexiflow.py:148]: typo

**Review by sonichi — APPROVED** [review]

**Review by jtongxin — APPROVED** [review]

**Pull request merged**